### PR TITLE
feat: add geocoding endpoints

### LINKED ISSUE #850: Use geoapify proxied by the backend

### PR DESCRIPTION
**Pull request by pyphilia**
close #850

todo: 
- [ ] Change secrets

### CODE REVIEW

File: src/services/item/plugins/geolocation/index.test.ts [hunk 1]
```diff
@@ -12,6 +13,8 @@ import { saveMember } from '../../../member/test/fixtures/members';
 import { savePublicItem } from '../../test/fixtures/items';
 import { ItemGeolocation } from './ItemGeolocation';
 
+jest.mock('node-fetch');
```
Comment (spaenleh) [inline src/services/item/plugins/geolocation/index.test.ts:16]: We don't use `nock` ?

File: src/utils/config.ts [hunk 2]
```diff
@@ -386,3 +386,11 @@ export const ALLOWED_SEARCH_LANGS = {
   fr: 'french',
   it: 'italian',
 };
+
+// Geolocation API Key
+if (!process.env.GEOLOCATION_API_KEY) {
```
Comment (spaenleh) [inline src/utils/config.ts:391]: Add in the env description in the Readme ? Also once stable it should be added to the docs. Will this be required when running the backend ? Can we provide any value if we do not plan to use that feature ?

File: src/services/item/plugins/geolocation/index.test.ts [hunk 3]
```diff
@@ -410,4 +413,162 @@ describe('Item Geolocation', () => {
       });
     });
   });
+
+  describe('GET /geolocation/reverse', () => {
+    beforeEach(() => {
+      (fetch as jest.MockedFunction<typeof fetch>).mockImplementation(async () => {
```
Comment (ReidyT) [inline src/services/item/plugins/geolocation/index.test.ts:419]: should this be in fixtures ?

**Review by spaenleh — APPROVED** [review]

**Review by ReidyT — APPROVED** [review]
LGTM 👍

File: src/services/item/plugins/geolocation/index.test.ts (hunk 1 above)
Comment (pyphilia) [reply src/services/item/plugins/geolocation/index.test.ts:16]: I never use nock, only alex did 🤷‍♀️

**Pull request merged**